File: 242. Valid Anagram/Valid_Anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        int i, ssize = s.size(), tsize = t.size();
        map<char,int>mci;
        if(ssize != tsize)
        {
            return false;
        }
        for(i = 0; i < ssize; i++)
        {
            mci[s[i]]++;
            mci[t[i]]--;
        }
        for(i = 0; i < ssize; i++)
        {
            if(mci[s[i]] != 0)
            {
                return false;
            }
        }
        return true;
    }
};

File: 242. Valid Anagram/Valid_Anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        
        mci = {}

        for i in range(len(s)):
            mci[s[i]] = mci.get(s[i], 0) + 1
            mci[t[i]] = mci.get(t[i], 0) - 1
        
        for i in range(len(s)):
            if mci[s[i]] != 0:
                return False
        
        return True
